scripts/validate_string_concatenation.py

```diff
@@ -102,28 +106,157 @@ def strings_to_concatenate(source_path: str) -> Generator[Tuple[str, int], None,
+    MSG: str = (
+        "String unnecessarily split in two by black. Please merge them manually."
+    )
+
     with open(source_path, "r") as file_name:
         tokens: List = list(tokenize.generate_tokens(file_name.readline))
 
     for current_token, next_token in zip(tokens, tokens[1:]):
         if current_token[0] == next_token[0] == token.STRING:
-            yield source_path, current_token[2][0]
+            yield source_path, current_token[2][0], MSG
+
+
+def strings_with_wrong_placed_space(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test case for leading spaces in concated strings.
+
+    For example:
+
+    >>> foo = (
+    ...    "bar "
+    ...    "baz"
+    ... )
+
+    Instead of:
+
+    >>> foo = (
+    ...        "bar"
+    ...       " baz"
+    ... )
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+    """
+    MSG: str = (
+        "String has a space at the beginning "
+        "instead of the end of the previous string."
+    )
+
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for first_token, second_token, third_token in zip(tokens, tokens[1:], tokens[2:]):
+        if (
+            first_token[0] == third_token[0] == token.STRING
+            and second_token[0] == token.NL
+        ):
+            # Means we are in a block of concated string
+
+            # Striping the quotes
+            first_string = first_token[1][1:-1]
+            second_string = third_token[1][1:-1]
+
+            if (not first_string.endswith(" ")) and (second_string.startswith(" ")):
+                yield source_path, third_token[2][0], MSG
+
+
+def bare_pytest_raises(source_path: str) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test Case for bare pytest raises.
+
+    For example:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    Instead of:
+
+    >>> with pytest.raise(ValueError, match="foo"):
+    ...     # Some code that raises ValueError
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #23922
+    """
+    MSG: str = "Bare pytests raise have been found."
+
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for counter, current_token in enumerate(tokens, start=1):
+        if current_token[0] == token.NAME and current_token[1] == "raises":
```

Comment:
Small preference, but I'd negate this condition and have a `continue` for it. So the rest of the code doesn't need to be nested. I think there are too many levels of indentation here.

Reply:
I don't see where the ```continue``` will be place in order to have less code indented.

Can you help me out?

Reply:
See these two cases:
```python
for item in items:
    if condition:
        do_stuff()
```
and
```python
for item in items:
    if condition:
        continue
    do_stuff()
```
If in the place of `do_stuff()` you've got lots of lines, the second is probably more readable, because you don't have a big block indented many lines, that is difficult to know exactly at which level it is.